core/modules/query/ValueExpr.h

```diff
@@ -91,8 +91,28 @@ class ValueExpr {
 
     bool addOp(query::ValueExpr::Op op);
 
+    /**
+     * @brief Get the Alias of the ValueExpr
+     */
     std::string const& getAlias() const { return _alias; }
+
+    /**
+     * @brief Set the Alias of the ValueExpr
+     *
+     * The state of the member variable `aliasIsUserDefined` will not be changed.
+     */
     void setAlias(std::string const& alias);
+
+    /**
+     * @brief Set the Alias of the ValueExpr, and indicate if the alias is user defined or not.
+     *
+     * The state of the member variable `aliasIsUserDefined` will be changed.
+     */
+    void setAlias(std::string const& alias, bool aliasIsUserDefined);
```

Comment:
I'm not sure how these two overloads of `setAlias` work w.r.t. `aliasIsUserDefined`. I think it's clear what happens if you call second overload - the state becomes either "user-defined" or "not-user-defined". But if you call first overload the state is "no-one-gives-a-damn"? Is that state useful? It's likely just my confusion as I do not know much about this code and I generally dislike modifiable state. 🙂